test/state/state.cpp

```diff
@@ -212,7 +212,9 @@ std::variant<TransactionReceipt, std::error_code> transition(State& state, const
     std::erase_if(state.get_accounts(),
         [](const std::pair<const address, Account>& p) noexcept { return p.second.destructed; });
 
-    auto receipt = TransactionReceipt{tx.type, result.status_code, gas_used, host.take_logs(), {}};
+    // Cumulative gas used is unknown in this scope.
```

Comment:
You should also set `.cumulative_gas_used` in this commit.
Is the `gas_used` a part of the receipt then?

Reply:
Done. `gas_used` is not a part of receipt in JsonRPC but it's a part of `TransactionReceipt` stuct. We need it there to pass used gas value out of `transition` function. We can separate it as return it as a additional return value but it would look bad to me.